### ISSUE TITLE
Allow "none" state serializer

### ISSUE THREAD

**Issue by endel** [colyseus/colyseus#335]
Currently, you either use `schema` or `fossil-delta` serializer. 

The problem is, if you do not use `this.setState()`, the `"fossil-delta"` is going to be used automatically. A few client implementations do not support `"fossil-delta"`, and end up throwing an error.

e.g. haxe:

```
{ [Error: FossilDelta serializer has been deprecated! Use SchemaSerializer instead.]
  val: 'FossilDelta serializer has been deprecated! Use SchemaSerializer instead.' }
```

So, introducing a `"none"` serializer, and checking it on every client should fix this.

**Comment by Gaen** [comment]
Why not just use `schema` by default?

Pros:
- conforms with deprecating `fossil-delta` yet still allows using it explicitly
- does not require introducing a stub serializer
- does not require touching clients (if I am not missing something here)

Cons?

**Comment by endel** [comment]
Hi @Gaen,

I'm afraid that using `"schema"` by default may cause issues if no state has been defined in some clients. 

- The C# client **requires** to define a generic `T` when joining a room (`.joinOrCreate<State>()`) if we don't have the `State`, we can't even call the method
- The Haxe client throws an error if the serializer is anything different than `schema` (https://github.com/colyseus/colyseus-hx/blob/master/src/io/colyseus/Room.hx#L130-L134), this one is easier to fix
- When the state is not set in the server, the server is printing warnings: https://github.com/colyseus/colyseus/blob/9c8722b932494f2256a9316ba1d56ce53f539507/src/Room.ts#L457-L460

I believe the easiest way to fix this would be to use a `"none"` serializer by default, and switch to `schema` in case a `Schema` structure is used in `.setState()`. If we can fix the clients to allow having `schema` + no state set it would be enough, though

**Comment by Gaen** [comment]
So let me rephrase a root problem:
- `colyseus` requires a room to have a state
- some users want to have a room without state
- it can be achieved on some clients by not calling `setState`
- this does not work on unity client because its interface explicitly requires state type parameter
- this does not work on haxe and other clients that do not support `fossil-delta`
- `fossil-delta` is deprecated and will be removed, which will completely break this workaround on all clients
- you would like to provide a clean official way of having a room without a state

Am I following you here?

> If we can fix the clients to allow having schema + no state set it would be enough, though

Here's a suggestion: provide an empty schema implementation for each client and be like "use this class if you don't want to have a state"

**Comment by endel** [comment]
As reported by @kevincastejon, in Unity we should be able to declare `Room` when no serializer is being used rather than the FossilDelta signature (`Room<IndexedDictionary<string, object>>`)

**Comment by endel** [comment]
The `NoneSerializer` has been added on 0.14!
